hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -191,6 +191,10 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         if (newTable
             && tbl.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP)
                 != null) {
+          if (tbl.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
```

Comment:
During [commitTransaction](https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/Transaction.java#L185)   there should be a way to validate if there is another content type and throw the same error.